wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -792,6 +935,57 @@ instance BuildableSafeGen NewAddress where
         newaddrAccountIndex
         newaddrWalletId
 
+data NewExternalAddress = NewExternalAddress
+  { neweaddrAccountIndex :: !AccountIndex
+  , neweaddrWalletId     :: !WalletId
+  } deriving (Show, Eq, Generic)
+
+deriveJSON Serokell.defaultOptions ''NewExternalAddress
+
+instance ToSchema NewExternalAddress where
+  declareNamedSchema =
+    genericSchemaDroppingPrefix "neweaddr" (\(--^) props -> props
+      & ("accountIndex" --^ "Target account's index to store this address in")
+      & ("walletId"     --^ "Corresponding external wallet identifier")
+    )
+
+instance Arbitrary NewExternalAddress where
+  arbitrary = NewExternalAddress <$> arbitrary
+                                 <*> arbitrary
+
+deriveSafeBuildable ''NewExternalAddress
+instance BuildableSafeGen NewExternalAddress where
+    buildSafeGen sl NewExternalAddress{..} = bprint("{"
+        %" accountIndex="%buildSafe sl
+        %" walletId="%buildSafe sl
+        %" }")
+        neweaddrAccountIndex
+        neweaddrWalletId
+
+-- | BIP44 derivation path, for work with external wallets, for example:
+-- m / purpose' / coin_type' / account' / change / address_index
+-- m /      44' /         0' /       0' /      0 /             1
+data AddressPath = AddressPath
+  { addrLevels :: ![Word32]
```

Comment:
Happy to see you settled for `Word32`, by skimming the BIP-44 proposal that seems exactly the size we need (4 bytes).